instrumentation/instaredigo/instaredigo_test.go

```diff
@@ -0,0 +1,419 @@
+// (c) Copyright IBM Corp. 2022
+
+//go:build go1.16
+// +build go1.16
+
+package instaredigo
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/gomodule/redigo/redis"
+	instana "github.com/instana/go-sensor"
+	"github.com/instana/testify/assert"
+	"github.com/instana/testify/require"
+)
+
+type MockConn struct {
+	address string
+}
+
+func (conn *MockConn) Do(commandName string, args ...interface{}) (reply interface{}, err error) {
+	reply = "OK <->" + commandName
+	if len(commandName) == 0 {
+		err = errors.New("Empty command received")
+	}
+	return reply, err
+}
+
+func (conn *MockConn) DoContext(ctx context.Context, commandName string,
+	args ...interface{}) (reply interface{}, err error) {
+	reply = "OK <->" + commandName
+	if len(commandName) == 0 {
+		err = errors.New("Empty command received")
+	}
+	return reply, err
+}
+
+func (conn *MockConn) ReceiveContext(ctx context.Context) (reply interface{}, err error) {
+	reply = "OK"
+	return reply, err
+}
+
+func (conn *MockConn) DoWithTimeout(timeOut time.Duration, commandName string,
+	args ...interface{}) (reply interface{}, err error) {
+	reply = "OK <->" + commandName
+	if len(commandName) == 0 {
+		err = errors.New("Empty command received")
+	}
+	return reply, err
+}
+
+func (conn *MockConn) ReceiveWithTimeout(timeout time.Duration) (reply interface{}, err error) {
+	reply = "OK"
+	return reply, err
+}
+
+func (conn *MockConn) Send(commandName string, args ...interface{}) error {
+	var err error
+	if len(commandName) == 0 {
+		err = errors.New("Empty command received")
+	}
+	return err
+}
+
+func (conn *MockConn) Receive() (reply interface{}, err error) {
+	reply = "OK"
+	return reply, err
+}
+
+func (conn *MockConn) Err() error {
+	err := errors.New("No error")
+	return err
+}
+
+func (conn *MockConn) Close() error {
+	err := errors.New("No error")
+	return err
+}
+
+func (conn *MockConn) Flush() error {
+	err := errors.New("No error")
+	return err
+}
+
+func TestMockDo(t *testing.T) {
```

Comment:
```suggestion
func TestDo(t *testing.T) {
```

Reply:
Because we are testing instrumented `Do`, not the mock's `Do`, it sounds more clear to me.

Reply:
Same for other test names.

Reply:
Agreed and done